base/bucket_gocb.go

```diff
@@ -1014,12 +1019,12 @@ func (bucket CouchbaseBucketGoCB) DeleteWithXattr(k string, xattrKey string) err
 		return err
 	}
 
-	_, err = bucket.Bucket.MutateInEx(k, gocb.SubdocDocFlagAccessDeleted, removeCas, 0).
+	_, deleteXattrErr := bucket.Bucket.MutateInEx(k, gocb.SubdocDocFlagAccessDeleted, removeCas, 0).
 		RemoveEx(xattrKey, gocb.SubdocFlagXattr).
 		Execute()
 
-	if err != nil && err != gocbcore.ErrSubDocSuccessDeleted {
-		return err
+	if deleteXattrErr != nil && deleteXattrErr != gocbcore.ErrSubDocSuccessDeleted {
```

Comment:
This is a little counterintuitive to have an error with "Success" in the name.  Can you add a comment to explain why `gocbcore.ErrSubDocSuccessDeleted` errors aren't treated as error conditions?